OpenRA.Mods.Common/Traits/Player/InsufficientFundsWarning.cs

```diff
@@ -0,0 +1,57 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Provides the player with an audible warning when their storage is nearing full.")]
+	public class InsufficientFundsWarningInfo : ITraitInfo, Requires<PlayerResourcesInfo>
+	{
+		[Desc("Interval (in ticks) at which to check if a warning should be played.")]
+		public readonly int WarnInterval = 750;
+
+		[Desc("The cash threshold below which a warning is played.")]
+		public readonly int Threshold = 50;
+
+		[Desc("The speech to play for the warning.")]
+		public readonly string Notification = "InsufficientFunds";
+
+		public object Create(ActorInitializer init) { return new InsufficientFundsWarning(init.Self, this); }
+	}
+
+	public class InsufficientFundsWarning : ITick
+	{
+		readonly InsufficientFundsWarningInfo info;
+		readonly PlayerResources resources;
+
+		int nextWarnTime = 0;
```

Comment:
This needs to be set to `info.WarnInterval` in the ctor otherwise the condition checking this will be true immediately.